x/gamm/keeper/pool.go

```diff
@@ -12,21 +12,22 @@ import (
 	"github.com/osmosis-labs/osmosis/v13/x/gamm/pool-models/balancer"
 	"github.com/osmosis-labs/osmosis/v13/x/gamm/pool-models/stableswap"
 	"github.com/osmosis-labs/osmosis/v13/x/gamm/types"
+	swaproutertypes "github.com/osmosis-labs/osmosis/v13/x/swaprouter/types"
 )
 
-func (k Keeper) MarshalPool(pool types.PoolI) ([]byte, error) {
+func (k Keeper) MarshalPool(pool swaproutertypes.PoolI) ([]byte, error) {
 	return k.cdc.MarshalInterface(pool)
 }
 
-func (k Keeper) UnmarshalPool(bz []byte) (types.PoolI, error) {
-	var acc types.PoolI
+func (k Keeper) UnmarshalPool(bz []byte) (types.CFMMPoolI, error) {
+	var acc types.CFMMPoolI
```

Comment:
Just to clarify, is it intended to Marshal from PoolI but Unmarshal into CFMMPoolI?

Reply:
Its unclear to me that its safe to Marshal to CFMMPoolI, I think we may have to always marshal and unmarshal from PoolI.

(Or at least need to double check the serialized bytecode. IIRC, the proto import path + struct name does get serialized into @type)

Reply:
I'm going to look into this.

Reply:
Please see test: [d7d388d](https://github.com/osmosis-labs/osmosis/pull/3560/commits/d7d388d30cf8d773f327d455d6e7c672b0cdad4d)

Reply:
Just noticed that you specified serializing specifically to JSON. Why would JSON matter? Would the test above cover your concern?

Reply:
JSON was suggested just for debugging